dev-libs/libuv/libuv-9999.ebuild

```diff
@@ -1,49 +1,67 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-inherit autotools git-r3 multilib-minimal
+
+inherit autotools multilib-minimal
 
 DESCRIPTION="Cross-platform asychronous I/O"
 HOMEPAGE="https://github.com/libuv/libuv"
-EGIT_REPO_URI="https://github.com/libuv/libuv"
+
+if [[ ${PV} = 9999* ]]; then
+	EGIT_REPO_URI="https://github.com/libuv/libuv.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/libuv/libuv/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~x64-cygwin ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+fi
 
 LICENSE="BSD BSD-2 ISC MIT"
 SLOT="0/1"
-KEYWORDS=""
-IUSE="static-libs"
-RESTRICT="test"
 
 BDEPEND="
 	sys-devel/libtool
-	virtual/pkgconfig
-"
+	virtual/pkgconfig"
 
 src_prepare() {
 	default
 
 	echo "m4_define([UV_EXTRA_AUTOMAKE_FLAGS], [serial-tests])" \
 		> m4/libuv-extra-automake-flags.m4 || die
 
+	if [[ ${CHOST} == *-darwin* && ${CHOST##*darwin} -le 9 ]] ; then
+		sed -i -e '/!defined(__DragonFly__)/i!defined(__APPLE__) && \\' \
+			src/unix/udp.c || die
+		sed -i -e '/err = pthread_setname_np/cerr = 0;' \
+			src/unix/darwin-proctitle.c || die
+		sed -i -e '/int uv_if_indextoname/i\
+		static size_t strnlen(char *x, size_t len) {\
+			char *r = memchr(x, '"'"'\\n'"'"', len);\
+			if (r == NULL)\
+				return len;\
+			else\
+				return r - x;\
+		}' \
+			src/unix/getaddrinfo.c || die
```

Comment:
This is ugly to say the least. Please turn it into a patch.

Reply:
Agreed, done